jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -425,11 +425,38 @@ def _extract_from_aperture(self, spectral_cube, uncert_cube, aperture,
                                            uncertainty=collapsed_as_mean.uncertainty,
                                            wcs=collapsed_as_mean.wcs,
                                            meta=collapsed_as_mean.meta)
+        elif selected_func == 'sum':
+            collapsed_nddata = getattr(nddata_reshaped, selected_func)(
+                axis=spatial_axes, **kwargs
+            )  # returns an NDDataArray
+            # Remove per steradian denominator
+            if astropy.units.sr in collapsed_nddata.unit.bases:
+                # Get pixel scale factor and apply units
+                pix_scale_factor = (self.spectral_cube.meta.get('PIXAR_SR', 1.0)
+                                    * (astropy.units.sr / astropy.units.pix))
+
+                # Get area of aperture with units
+                aperture_area_with_units = self.aperture_area_along_spectral * astropy.units.pix
+
+                # Multiply all units together to get correct sum units
+                new_unit = (collapsed_nddata.unit * aperture_area_with_units.unit
+                            * pix_scale_factor.unit)
+                collapsed_nddata.unit = new_unit
+
+                # Data cannot be set be can be modified in place
+                collapsed_nddata.data[:] = (collapsed_nddata.data
+                                            * self.aperture_area_along_spectral)
+
+                # Update uncertainty units as well
+                if hasattr(collapsed_nddata.uncertainty, 'unit'):
+                    new_uncert_unit = (collapsed_nddata.uncertainty.unit
+                                       * aperture_area_with_units.unit
+                                       * pix_scale_factor.unit)
+                    collapsed_nddata.uncertainty.unit = new_uncert_unit
```

Comment:
The uncertainties themselves should probably be integrated as well, and I think this can be simplified quite a bit to something like (note: I have not actually tested this and this hardcodes sr, but that is probably ok for now since we're accessing `PIXAR_SR` 🤷‍♂️ ):

```suggestion
                aperture_area = self.aperture_area_along_spectral * self.spectral_cube.meta.get('PIXAR_SR', 1.0) * u.sr
                collapsed_nddata.unit *= aperture_area.unit
                collapsed_nddata.data[:] *= aperture_area
                
                # Update uncertainty units as well
                 collapsed_nddata.uncertainty[:] *= aperture_area
                 if hasattr(collapsed_nddata.uncertainty, 'unit'):
                    collapsed_nddata.uncertainty.unit *= aperture_area.unit
```



It might even be possible to simplify this further and multiply the quantities themselves (`collapsed_data[:] *= aperture_area`, etc), but I'm not sure 100% if the nddata object here supports that or not.

Reply:
were the tests passing before?  Perhaps something is missing here - I didn't actually test the code suggestions, just tried to make the point that I think we can create a single quantity that is the integral over the aperture in sky units (steradians) and multiply by that directly.  I would expect these changes to possibly affect the tests you wrote, since PIXAR_SR was not previously in the factor that was multiplying the actual values, but I wouldn't expect other tests to fail.